src/compas_fab/backends/ros/backend_features/move_it_iter_inverse_kinematics.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from compas.geometry import Sphere
+
+from compas_fab.backends import BackendError
+from compas_fab.backends.interfaces import IterInverseKinematics
+from compas_fab.robots.constraints import PositionConstraint
+
+
+class MoveItIterInverseKinematics(IterInverseKinematics):
+    """Callable to iterate calculation of the robot's inverse kinematics for a given frame."""
+    def __init__(self, ros_client):
+        self.ros_client = ros_client
+
+    def iter_inverse_kinematics(self, robot, frame_WCF, start_configuration=None, group=None, options=None):
```

Comment:
one question remains open a bit if a user should expect `iter_inverse_kinematics` to return unique solutions.
in the case of analytic ik and 6-ax robot we can be sure to get at most 8, in the case of using moveit kdl and a robot on a linear axis we get infinite, and with moveit kdl only on a 6-ax-robot we could potentially get several close ones...

Reply:
I don't think that a user should expect that... As you suggest, each use case has a different set of outcomes, which can be influenced by the constraints a user provides.  I'm not sure how within the function call it would be known which of those various scenarios we are operating to thereby stop yielding configurations at the appropriate time.  And if you cached configurations within the generator to only yield unique configurations, I think it would be very easy to create an infinite (or by all appearances infinite) loop trying to find the next one.  I think we should expect the user to use heuristics about their particular robotic set-up, and use the iterator appropriately.